src/repo.rs

```diff
@@ -566,13 +570,19 @@ impl DocumentInfo {
         document: Arc<RwLock<SharedDocument>>,
         handle_count: Arc<AtomicUsize>,
     ) -> Self {
+        let last_heads = {
+            let doc = document.read();
+            doc.automerge.get_heads()
+        };
         DocumentInfo {
             state,
             document,
             handle_count,
             sync_states: Default::default(),
             change_observers: Default::default(),
-            patches_since_last_save: 0,
+            patches_since_last_compact: 0,
```

Comment:
I think this is a confusing name because patches in `automerge` are a different thing, they're the changes needed to update a materialized view. Is there a reason we can't use `changes_since_last_compact`?